components/next_steps_view/steps/team_profile_step/team_profile_step.tsx

```diff
@@ -0,0 +1,218 @@
+import Input from 'components/input';
+import PictureSelector from 'components/picture_selector';
+import {AcceptedProfileImageTypes} from 'utils/constants';
+import * as Utils from 'utils/utils';
+
+import {StepComponentProps} from '../../steps';
+
+import './team_profile_step.scss';
+
+type Props = StepComponentProps & {
+    team: Team & {last_team_icon_update?: number};
+    siteURL: string;
+    maxFileSize: number;
+    actions: {
+        patchTeam: (team: Team) => void;
+        removeTeamIcon: (teamId: string) => void;
+        setTeamIcon: (teamId: string, imageData: File) => void;
+    };
+};
+
+type State = {
+    teamName: string;
+    teamNameError?: string;
+    teamURL: string;
+    teamURLError?: string;
+    profilePicture?: File;
+    profilePictureError: boolean;
+    removeProfilePicture: boolean;
+};
+
+export default class TeamProfileStep extends React.PureComponent<Props, State> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            teamName: props.team.display_name,
+            teamURL: props.team.name,
+            profilePictureError: false,
+            removeProfilePicture: false,
+        };
+    }
+
+    private handleNameInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        let teamNameError;
+        if (!event.target.value) {
+            teamNameError = Utils.localizeMessage('next_steps_view.team_profile_step.nameCannotBeBlank', 'Team name can’t be blank');
+        }
+
+        this.setState({teamName: event.target.value, teamNameError});
+    }
+
+    private handleURLInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        let teamURLError;
+        if (!event.target.value) {
+            teamURLError = Utils.localizeMessage('next_steps_view.team_profile_step.nameCannotBeBlank', 'Team URL can’t be blank');
+        }
+
+        const urlRegex = /^[a-z]+([a-z\-0-9]+|(__)?)[a-z0-9]+$/g;
+        if (!urlRegex.test(event.target.value)) {
+            teamURLError = Utils.localizeMessage('next_steps_view.team_profile_step.malformedURL', 'Use only lower case letters, numbers and dashes. Must start with a letter and can\'t end in a dash.');
+        }
+
+        this.setState({teamURL: event.target.value, teamURLError});
+    }
+
+    isFinishDisabled = () => {
+        return Boolean(!this.state.teamName || this.state.teamNameError || !this.state.teamURL || this.state.teamURLError || this.state.profilePictureError);
+    }
+
+    getSanitizedSiteURL = () => {
+        const removeProtocol = this.props.siteURL.replace(/(^\w+:|^)\/\//, '');
+        if (removeProtocol[removeProtocol.length - 1] === '/') {
+            return removeProtocol;
+        }
+
+        return `${removeProtocol}/`;
+    }
+
+    onSkip = () => {
+        this.props.onSkip(this.props.id);
+    }
+
+    onFinish = () => {
+        const team = Object.assign({}, this.props.team, {display_name: this.state.teamName});
+
+        this.props.actions.patchTeam(team);
```

Comment:
no need for await here?

Reply:
Nope, we can fire and forget. There's no component state that depends on the outcome.